### PR TITLE
Migrate some files to typescript

### PR DESCRIPTION
**Pull request by anuraaga**
Will gradually migrate some files at a time. Figured it's especially good to have a first one to work out kinks and have an example.

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
works for me, but leaving this up to you and @tacigar

File: zipkin-lens/src/components/App/SidebarMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  }),
+);
+
+interface SidebarMenuProps extends React.DOMAttributes<HTMLElement> {
+  title: string;
+  path: string;
+  icon: IconProp;
+}
+
+export const SidebarMenuImpl = React.forwardRef<
```
Comment (tacigar) [inline zipkin-lens/src/components/App/SidebarMenu.tsx:53]: just curious.
why is this component made forwardRef?

forwardRef seems to be done in this commit.

https://github.com/openzipkin/zipkin/commit/7a1790f3b3e61110cd4e7263219702d398b64efc#diff-d7507c88e4d135f78935a54009f7fefd

Comment (anuraaga) [reply zipkin-lens/src/components/App/SidebarMenu.tsx:53]: Needed to pass a ref to the DOM element to position the popup menu

https://github.com/openzipkin/zipkin/blob/master/zipkin-lens/src/components/App/LanguageSelector.jsx#L85

Pretty sure I tried without ref and wasn't being able to position the popup menu well, let me know if you have any alternate ideas though.

Comment (tacigar) [reply zipkin-lens/src/components/App/SidebarMenu.tsx:53]: I get it! thanks

**Review by tacigar — APPROVED** [review]
awesome! thanks!

**Pull request merged**